src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,549 @@
+                    with open(arc_agent_container_logs, 'w+') as container_file:
+                        container_file.write(str(container_log))
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+
+    except Exception as e:
+        print(e)
+    
+    return storage_space_available
+
+
+def arc_agents_event_logger(filepath_with_timestamp, storage_space_available):
+
+    try:
+
+        if storage_space_available :
+
+            # CMD command to get helm values in azure arc and converting it to json format
+            command = ["kubectl", "get", "events", "-n", "azure-arc", "--output", "json"]
+
+            # Using subprocess to execute the helm get values command and fetching the output
+            p = subprocess.Popen(command, stdout=subprocess.PIPE)
+            text = p.stdout.read()
+            
+            # Converting output obtained in json format and fetching the clusterconnect-agent feature
+            my_json = json.loads(text)
+
+            event_logs_path=os.path.join(filepath_with_timestamp,"Arc_Agents_Events.txt")
+            with open(event_logs_path, 'w+') as event_log:
+
+                for events in my_json["items"] :
+                        event_log.write(str(events) + "\n")
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+            
+    except Exception as e:
+        print(e)
+    
+    return storage_space_available
+
+
+def deployments_logger(appv1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    try:
+        if storage_space_available :
+
+            # Creating new Deployment Logs folder in the given timestamp folder
+            deployments_path = os.path.join(filepath_with_timestamp, "Deployment_logs")
+            try:
+                os.mkdir(deployments_path)
+            except FileExistsError:
+                pass
+
+            # To retrieve all the the deployements that are present in the Cluster
+            deployments_list = appv1_api_instance.list_namespaced_deployment("azure-arc")
+
+            # Traversing through all the deployments present
+            for deployment in deployments_list.items:
+
+                # Fetching the deployment name
+                deployment_name = deployment.metadata.name
+
+                deployment_logs_path = os.path.join(deployments_path, deployment_name + ".txt")
+                # Creating a text file with the name of the deployment and adding deployment status in it
+                with open(deployment_logs_path, 'w+') as deployment_file:
+                    deployment_file.write(str(deployment.status))
+
+    except OSError as e:
+        storage_space_available = False
+        shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+            
+    except Exception as e:
+        print(e)
```

Comment:
same comment as above. Use logger everywhere and add some description before printing the exception directly. Example - https://github.com/Azure/azure-cli-extensions/blob/a31daa97a1bfb02d48e09418c493eeda8ead24b6/src/connectedk8s/azext_connectedk8s/custom.py#L234